shared/origami/CMakeLists.txt

```
@@ -0,0 +1,156 @@
+option(ORIGAMI_ENABLE_PYTHON "Enable Python bindings." OFF)
+option(ORIGAMI_BUILD_TESTING "Enable Python binding tests." OFF)
+
+find_package(hip REQUIRED)
+
+if(ORIGAMI_BUILD_SHARED_LIBS)
+    add_library(origami SHARED) 
+else()
+    add_library(origami STATIC)
+endif()
+
+rocm_set_soversion(origami ${PROJECT_VERSION_MAJOR})
+
+add_library(origami::origami ALIAS origami)
+
+set_target_properties(origami PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_features(origami PUBLIC cxx_std_17)
+
+target_include_directories(origami
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+target_sources(origami
+    PRIVATE
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/gemm.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/hardware.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/utils.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/streamk.cpp"
+)
+
+target_link_libraries(origami PUBLIC hip::host)
+
+if(ORIGAMI_ENABLE_PYTHON)
+    find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+    find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
+
+    include(FetchContent)
+    FetchContent_Declare(
+        nanobind
+        GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+        GIT_TAG        v1.9.2  # or latest stable version
+    )
+    FetchContent_MakeAvailable(nanobind)
+    find_package(nanobind CONFIG REQUIRED)
+
+    nanobind_add_module(origami_python 
+        "${CMAKE_CURRENT_SOURCE_DIR}/python/origami_module.cpp"
+    )
+    
+    set_target_properties(origami_python PROPERTIES
+        OUTPUT_NAME "origami"
+    )
+    
+    target_link_libraries(origami_python PRIVATE origami::origami)
+endif()
+
+
+if(ORIGAMI_BUILD_TESTING OR BUILD_TESTING)
+    enable_testing()
+    
+    if(ORIGAMI_ENABLE_PYTHON)
+        add_test(
+            NAME origami_python_basic_test
+            COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/python/origami_test.py
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/python
+        )
+        
+        add_test(
+            NAME origami_python_grid_test  
+            COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/python/origami_grid_test.py
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/python
+        )
+
+        set_tests_properties(origami_python_basic_test origami_python_grid_test
+            PROPERTIES
+                ENVIRONMENT "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}:${CMAKE_CURRENT_SOURCE_DIR}/python:$ENV{PYTHONPATH}"
+        )
+
+        set_tests_properties(origami_python_basic_test origami_python_grid_test
+            PROPERTIES
+                DEPENDS origami_python
+        )
+    endif()
+endif()
+
+
+if(ORIGAMI_STANDALONE)
+    rocm_install(TARGETS origami)
+
+    rocm_export_targets(
+        TARGETS origami::origami
+        DEPENDS PACKAGE hip
+        NAMESPACE origami::
```

```suggestion
        NAMESPACE roc:
```

see discussion above